### PR TITLE
N muartem 

### PR DESCRIPTION
**Pull request by muartem**
https://user-images.githubusercontent.com/25155242/108938382-a9a08100-7658-11eb-81b4-3c9ba9f4a7ce.mp4

### CODE REVIEW

File: src/js/UI.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    static modalClose = document.querySelectorAll('.modal-close-btn');
+
+    static init() {
+        const { loginButton, signUpButton, openLogin, openSignup, loginModal,
+            signUpModal, modalClose, loginReDir, signupReDir } = UI;
+
+        [openLogin, loginReDir].forEach(btn => btn.addEventListener('click', () => {
+            loginModal.style.display = 'block';
+            signUpModal.style.display = 'none';
```
Comment (alexgul1) [inline src/js/UI.js]:
```suggestion
            loginModal.classList.add('active');
            signUpModal.classList.remove('active');
```

File: src/js/UI.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import SignIn from './SignInUI';
+import SignUp from './SignUpUI';
+
+export default class UI{
```
Comment (matroskin062) [inline src/js/UI.js]: I think it's not quite correct class name in this case

**Review by matroskin062 — CHANGES_REQUESTED** [review]

**Comment by matroskin062** [comment]
Please add check if User alerady logged in, because login from renders each time when we are loading page

File: src/js/SignInUI.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  static signIn() {
+    const { user, pass, error, errorLog} = SignIn;
+    if (user.value.length < 1 || pass.value.length < 1) {
+      errorLog(`There is an empty field`, error)
+    } else {
+      AuthAPI.login({
+        identifier: user.value,
+        password: pass.value,
+      })
+        .then(() => console.log('sign in')) // Передать даннные на UserUI
```
Comment (matroskin062) [inline src/js/SignInUI.js]: what is the purpose of this `then` method?

Comment (muartem) [reply src/js/SignInUI.js]: plug

File: src/js/AuthUI.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+        [openSignup, signupReDir].forEach(btn => btn.addEventListener('click', () => {
+            loginModal.classList.contains('active') && loginModal.classList.remove('active');
+            signUpModal.classList.add('active');
+        }));
+
+        modalClose.forEach(btn => {
+            btn.addEventListener('click', () => {
+                loginModal.classList.contains('active') && loginModal.classList.remove('active');
+                signUpModal.classList.contains('active') && signUpModal.classList.remove('active');
```
Comment (alexgul1) [inline src/js/AuthUI.js]:
```suggestion
            signUpModal.classList.remove('active');
        }));

        [openSignup, signupReDir].forEach(btn => btn.addEventListener('click', () => {
            loginModal.classList.remove('active');
            signUpModal.classList.add('active');
        }));

        modalClose.forEach(btn => {
            btn.addEventListener('click', () => {
                loginModal.classList.remove('active');
                signUpModal.classList.remove('active');
```

`classList.remove` will not fail if there is no such class

File: src/js/SignInUI.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  signIn() {
+    const { user, pass, idError, passError, errorLog} = this;
+    let errors = 0;
+    if (user.value.length < 1) {
+      errorLog(`There is an empty Username`, idError);
+      errors++;
+    }if (pass.value.length < 1) {
+      errorLog(`There is an empty Password`, passError);
+      errors++;
```
Comment (alexgul1) [inline src/js/SignInUI.js]:
```suggestion
    if (!user.value.trim()) {
      errorLog(`There is an empty Username`, idError);
      errors++;
    }if (!pass.value.trim()) {
      errorLog(`There is an empty Password`, passError);
      errors++;
```

there was no check for spaces `"     "`

File: src/js/SignUpUI.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    const { user, pass, email, userError, emailError, signupError, errorLog} = this;
+    const  regExpEmail = new RegExp(
+        /^([a-zA-Z0-9_\-.]+)@([a-zA-Z0-9_\-.]+)\.([a-zA-Z]{2,5})$/
+    );
+    let errors = 0;
+    if (user.value.length < 1) {
+      errorLog(`There is an empty field`, userError)
+      errors++;
+    }
+    if (pass.value.length < 1 ) {
```
Comment (alexgul1) [inline src/js/SignUpUI.js]:
```suggestion
    if (!user.value.trim()) {
      errorLog(`There is an empty field`, userError)
      errors++;
    }
    if (!pass.value.trim()) {
```

File: src/js/SignUpUI.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    if (!regExpEmail.test(email.value)) {
+      errorLog(`Invalid email example@test.io`, emailError)
+      errors++;
+    }
+    if (errors === 0){
+      AuthAPI.registration({
+        username: user.value,
+        email: email.value,
+        password: pass.value,
```
Comment (alexgul1) [inline src/js/SignUpUI.js]:
```suggestion
        username: user.value.trim(),
        email: email.value.trim(),
        password: pass.value.trim(),
```
we should trim the sent data

File: src/js/SignInUI.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    }
+    if (errors === 0) {
+      AuthAPI.login({
+        identifier: user.value,
+        password: pass.value,
```
Comment (alexgul1) [inline src/js/SignInUI.js]:
```suggestion
        identifier: user.value.trim(),
        password: pass.value.trim(),
```

File: src/js/SignInUI.js (hunk 5 above)
Comment (matroskin062) [reply src/js/SignInUI.js]: In addition to this pls fix code formatting

File: package.json [hunk 9]
```diff
@@ -27,15 +29,17 @@
...
   "dependencies": {
     "@babel/runtime": "^7.12.18",
-    "axios": "^0.21.1"
+    "axios": "^0.21.1",
+    "reset.scss": "^1.0.0"
   },
   "husky": {
     "hooks": {
-      "pre-commit": "lint-staged"
+
```
Comment (matroskin062) [inline package.json]:
```suggestion
"pre-commit": "lint-staged"
```

**Review by matroskin062 — CHANGES_REQUESTED** [review]

**Comment by alexgul1** [comment]
refactoring some files

**Review by matroskin062 — APPROVED** [review]

**Review by alexgul1 — APPROVED** [review]

**Pull request merged**